### PR TITLE
Issue #1026: Add config= gate to observation AC condition check

### LINKED ISSUE #1026: observation: プロジェクト設定依存の observation AC に .wholework.yml 設定有効性の事前チェックゲートを追加
## Background

`verify-type: observation event=auto-run` は「任意の `/auto` 完走」という広いイベントに対して発火するため、観察条件が特定のプロジェクト設定に依存する場合、その設定が対象リポジトリで無効だと条件が原理的に観察不能であるにもかかわらず、notification コメントだけが繰り返し蓄積する。

実測 (2026-07-20 の `/auto 1025` 完走後の L3 observation dispatch):

- #797 (`always-pr: true` での demotion 抑止観察) — 本リポジトリでは `always-pr` 未設定のため観察不能。2026-06-28 以降、observation notification コメントが 30 件超蓄積し、`/verify` 再実行 4 回すべてで SKIPPED
- 同種のパターンは #783 (`auto-stop-at: review` の観察) 等、プロジェクト設定依存の observation AC 全般に及ぶ

既存のゲート機構として `opportunistic-search.sh` の `--context-file` + `keyword=` 属性 (AC 行のキーワードが context file に出現する場合のみマッチ) があるが、`.wholework.yml` の設定有効性を事前チェックする機構は存在しない。

検出元: #797 の Verify Retrospective (Tier 2 memory proposal として Spec `docs/spec/issue-797-always-pr-step3a-demotion.md` に記録済み)。

## Purpose

プロジェクト設定依存の observation AC に「本プロジェクトの `.wholework.yml` で該当設定が有効か」を事前チェックするゲートを追加し、原理的に観察不能な条件への notification コメント蓄積と `/verify` dispatch の空振りを削減する。

## Proposal (Outline)

- **A. AC 側マーカーの拡張**: observation AC に設定前提を宣言できる属性を追加する (例: `<!-- verify-type: observation event=auto-run config=always-pr -->`)。既存の `keyword=` 属性と同型の拡張
- **B. ゲート判定の実装**: `opportunistic-search.sh` (または `observation-trigger.sh`) が `config=<key>` 属性を持つ AC について、対象リポジトリの `.wholework.yml` で該当 key が有効かをチェックし、無効なら match から除外する (notification コメント投稿と dispatch の両方を抑止)
- **C. 後方互換**: `config=` 属性なしの observation AC は従来どおり無条件マッチ (既存 Issue への影響なし)
- **D. 既存 Issue への適用**: #797 等、設定依存が判明している既存 observation AC への `config=` 属性の追記は本 Issue のスコープに含めるか `/spec` フェーズで判断

実装方式の最終判断 (属性名・判定の実装箇所・`capabilities.*` ネスト key の扱い等) は `/spec` フェーズで行う。

### Pre-merge (auto-verified)

- [x] observation AC に設定前提を宣言する属性 (例: `config=<key>`) の仕様が定義され、ゲート判定が実装されている 
- [x]  `config=` 属性の仕様が `modules/observation-trigger.md` にドキュメント化されている (先例 #934 の `keyword=` セクションと同様の追記形式を想定)
- [x] 属性なしの observation AC は従来どおり無条件マッチする (後方互換) 
- [x] ゲート判定のテストが追加されている (設定有効時マッチ / 無効時除外 / 属性なし無条件マッチの 3 ケース) 
- [x]  `config=` ゲートのテストケースが `tests/opportunistic-search.bats` に追加されている

### Post-merge

- [x] 設定無効なプロジェクトで該当 observation AC への notification コメントが蓄積しなくなることを観察 

## Related Issues

- 検出元: #797 (Verify Retrospective — Spec に Tier 2 memory proposal として記録済み)
- 直接の先例: #934 (`keyword=` 条件チェックゲートの実装。`opportunistic-search.sh` の event モードマッチングループに、AC 行から属性を抽出して外部コンテキストと照合するゲートを追加した過去実装。`grep` ベースの行内属性抽出 + 属性なし時は無条件マッチという後方互換設計は、本 Issue の `config=` ゲートでもそのまま踏襲できる)
- 同種パターン: #783 (`auto-stop-at: review` の観察)、#839 / #843 / #984 / #995 / #1009 (route・recovery 発火依存 — 設定依存とは別クラスだが notification 蓄積の様相は同一)
- 関連 (別機構): #952 (event-based observation scan の dispatch fan-out 制御 — dispatch 量の制御。本 Issue は観察可能性の事前ゲートで相補関係)
- 既存の先例機構: `opportunistic-search.sh` の `--context-file` + `keyword=` 属性ゲート (#934 で実装)

## Autonomous Auto-Resolve Log

- **[ゲート判定の一次実装箇所は `opportunistic-search.sh` を軸とする想定 (先例 #934 踏襲)]** — reason: #934 で `keyword=` 属性ゲートが `opportunistic-search.sh` の event モードマッチングループ (行単位の属性抽出 → 条件不一致なら `continue`) に実装済みで、`config=` ゲートも同じ抽出・除外の仕組みをそのまま延長できる。ただし `keyword=` は呼び出し元が用意する外部コンテキスト (`--context-file`) との照合が必要だったのに対し、`config=` は `.wholework.yml` をリポジトリルートから直接読めるため、`observation-trigger.sh` 側に新たな CLI 引数 (`--context-file` 相当) を追加する必然性は低いと考えられる。最終的な実装箇所・引数設計の確定は `/spec` フェーズの判断に委ねる (Issue 本文にも明記済み)。
  - Other candidates: `observation-trigger.sh` 側で `.wholework.yml` を読み込んで `opportunistic-search.sh` に `--config-valid`/`--config-invalid` 相当のフラグを渡す設計 (呼び出し階層が増えるため、既存の `keyword=` 実装との対称性は低い)

### PR DESCRIPTION
**Pull request by saito**
## Summary
- `scripts/opportunistic-search.sh` の event モードマッチングループに `config=<key>` 属性ゲートを追加。マッチした observation AC 行が `config=<key>` を持つ場合、`.wholework.yml` の該当キーが `true`（大小文字非依存）でなければ match から除外する。新規 CLI 引数は追加せず、既存の `scripts/get-config-value.sh` を経由して `.wholework.yml` を CWD 相対で読む
- `modules/observation-trigger.md` に既存の `keyword=` ゲート節と同型の `## Condition Check Gate (config=)` 節を追加し、属性書式・判定仕様・スコープ制約（フラット kebab-case キーのみ、真偽値のみ）をドキュメント化
- `tests/opportunistic-search.bats` に `config=` ゲートの 3 ケース（有効時マッチ / 無効時除外 / 属性なし無条件マッチ）を追加

## Verification (pre-merge)
- observation AC の設定前提宣言属性の仕様定義とゲート判定実装
- `config=` 属性の仕様が `modules/observation-trigger.md` にドキュメント化されている
- 属性なし AC の無条件マッチ（後方互換）が確認できる
- ゲート判定のテスト（3 ケース）が追加されている
- `config=` ゲートのテストケースが `tests/opportunistic-search.bats` に追加されている

## Verification (post-merge)
- 設定無効なプロジェクトで該当 observation AC への notification コメントが蓄積しなくなることを観察 (observation event=auto-run)

closes #1026

### CODE REVIEW

File: tests/opportunistic-search.bats [hunk 1]
```diff
@@ -227,6 +227,42 @@ teardown() {
...
+    export MOCK_ISSUE_BODY_601='- [ ] Verify demotion is suppressed <!-- verify-type: observation event=auto-run config=some-flag -->'
+    export WHOLEWORK_CONFIG_PATH=/dev/null
+
+    run bash "$SCRIPT" --event auto-run
+    unset WHOLEWORK_CONFIG_PATH
+    [ "$status" -eq 0 ]
+    [ "$output" = "[]" ]
+}
+
+@test "config gate: AC without config= matches unconditionally" {
```
Comment (saito) [inline tests/opportunistic-search.bats:254]: [Edge Case/Robustness] **CONSIDER**: The `keyword=` gate has a dedicated regression test covering the `sed -e 's/-*$//'` trailing-dash-stripping edge case (AC line written without a space before `-->`). The new `config=` gate reuses the identical extraction pattern (`grep -oE 'config=[^ >]+' | sed -e 's/^config=//' -e 's/-*$//'`) but has no equivalent test case for the no-space-before-`-->` scenario. Risk is low since the code path is shared and already exercised by the keyword test, but a config-specific regression would not be caught if the two extraction blocks ever diverge.

Recommended fix: optionally add a fourth `config=` test case mirroring the existing keyword "no space before -->" test, e.g. `config=some-flag-->` with `some-flag: true` in the fixture, asserting the match still succeeds.

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| Condition | Result | Notes |
|-----------|--------|-------|
| observation AC の設定前提宣言属性の仕様定義とゲート判定実装 | PASS | `modules/observation-trigger.md` § Condition Check Gate (`config=`) に仕様定義、`scripts/opportunistic-search.sh` にゲート判定実装を確認 |
| `config=` 属性の仕様が `modules/observation-trigger.md` にドキュメント化 | PASS | `file_contains` 検証済み |
| 属性なしの observation AC は従来どおり無条件マッチ (後方互換) | PASS | 実装 (`CONFIG_KEY` 未抽出時は分岐スキップ) とテスト (`config gate: AC without config= matches unconditionally`) で確認 |
| ゲート判定のテストが追加 (設定有効時マッチ / 無効時除外 / 属性なし無条件マッチの3ケース) | PASS | `tests/opportunistic-search.bats` に3ケースとも存在、実行して25/25 PASS を確認 |
| `config=` ゲートのテストケースが `tests/opportunistic-search.bats` に追加 | PASS | `file_contains` 検証済み |
| (Post-merge) 設定無効なプロジェクトで notification コメントが蓄積しなくなることを観察 | POST-MERGE | merge 後、次回 `/auto` 完走時の observation dispatch で観察 |

## CI Status

| Job name | Status | Notes |
|---------|--------|-------|
| DCO | SUCCESS | — |
| Run bats tests | SUCCESS | `tests/opportunistic-search.bats` 25/25 PASS (新規 `config=` ゲート3ケース含む)、`tests/get-config-value.bats`・`tests/observation-trigger.bats` も回帰なしをローカルで再確認 |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## Code Review

review-light エージェントによる4観点統合レビュー (Spec乖離 / エッジケース・堅牢性 / セキュリティ / ドキュメント整合性) を実施。

### Issue Count Summary

- Posted as line comments: 1 (MUST: 0, SHOULD: 0, CONSIDER: 1)
- General Comments (in body): 0

---
Next: run `/merge 1030`.

**Comment by saito** [comment]
## Review Response Summary

### Claude Review Response

| Aspect | Filename | Line | Issue | Response | Fix/Skip reason |
|--------|---------|------|-------|----------|----------------|
| エッジケース・堅牢性 | tests/opportunistic-search.bats | 254 | `config=` ゲートに `keyword=` の「`-->` 前にスペースなし」相当の回帰テストがない | Skipped | CONSIDER (任意対応)。抽出ロジック (`grep -oE '...[^ >]+' | sed -e 's/-*$//'`) は `keyword=` ゲートと共有コードパスであり、同シナリオは既存の `keyword=` 用テストで既にカバー済み。リスクは低いと判断し、本PRでの追加は見送り |

### Lightweight Re-check Results

- Tests: PASS (`bats tests/opportunistic-search.bats` 25/25、`tests/get-config-value.bats`・`tests/observation-trigger.bats` も回帰なしをローカルで確認)
- Fix diff check: N/A (修正コミットなし)
- New issues: none

---
Next: run `/merge 1030`.
